### PR TITLE
Added basic relations to `Data.Bool`

### PR DESCRIPTION
**Pull request by MatthewDaggitt**
Trying again. Recently needed to define a lexicographic ordering over Data.Fin.Subset and realised that these were missing.

### CODE REVIEW

File: src/Data/Bool/Properties.agda [hunk 1]
```diff
@@ -51,6 +51,170 @@ false ≟ true  = no λ()
...
+≤-refl = ≤-reflexive refl
+
+≤-antisym : Antisymmetric _≡_ _≤_
+≤-antisym b≤b _ = refl
+
+≤-trans : Transitive _≤_
+≤-trans f≤t b≤b = f≤t
+≤-trans b≤b f≤t = f≤t
+≤-trans b≤b b≤b = b≤b
+
```
Comment (gallais) [inline src/Data/Bool/Properties.agda]: I would write a lazier version instead:

```agda
≤-trans : Transitive _≤_
≤-trans b≤b p   = p
≤-trans f≤t b≤b = f≤t
```

File: src/Data/Bool/Properties.agda [hunk 2]
```diff
@@ -51,6 +51,170 @@ false ≟ true  = no λ()
...
+≤-total : Total _≤_
+≤-total false false = inj₁ b≤b
+≤-total false true  = inj₁ f≤t
+≤-total true  false = inj₂ f≤t
+≤-total true  true  = inj₁ b≤b
+
```
Comment (gallais) [inline src/Data/Bool/Properties.agda]: I'd hoist `≤-minimum` and `≤-maximum` above `≤-total` and write:
```agda
≤-total : Total _≤_
≤-total false b = inj₁ (≤-minimum b)
≤-total true  b = inj₂ (≤-maximum b)
```

File: src/Data/Bool/Properties.agda [hunk 3]
```diff
@@ -51,6 +51,170 @@ false ≟ true  = no λ()
...
+infix 4 _≤?_
+
+_≤?_ : Decidable _≤_
+false ≤? false = yes b≤b
+false ≤? true  = yes f≤t
+true  ≤? false = no λ()
+true  ≤? true  = yes b≤b
+
```
Comment (gallais) [inline src/Data/Bool/Properties.agda]: Same here:

```agda
_≤?_ : Decidable _≤_
false ≤? b     = yes (≤-minimum b)
true  ≤? false = no λ ()
true  ≤? true  = yes b≤b
```

**Comment by MatthewDaggitt** [comment]
Nice spots all of them. Have to confess I brute forced most of this using `auto`.

**Pull request merged**